virttest/qemu_devices/qcontainer.py

```diff
@@ -42,19 +42,22 @@ class DevContainer(object):
     """
     # General methods
 
-    def __init__(self, qemu_binary, vmname, strict_mode="no",
+    def __init__(self, qemu_binary, vmname, machine_type, strict_mode="no",
```

Comment:
Always add new arguments to the end of the list to avoid breaking existing calls... What I mean is someone might be using `DevContainer(bin, name, "yes")` directly (and there is at least one test `io-github-autotest-qemu/qemu/tests/slof_device_tree.py`) that would be affected by your change. When you introduce the new argument as last providing a safe default, the change will be backward compatible. The question is what is backward compatible as s390 does not define `virt` and arm won't work without `-M`...

Reply:
Actually we could perhaps use `machine_type=none`. Not sure whether we use any features that would be affected.